### ISSUE TITLE
API Key not Valid

### ISSUE THREAD

**Issue by JosuaWelss** [AwesomeTTS/awesometts-anki-addon#258]
I setup Microsoft Azure with Azure student.

I tried using it with AwesomeTTS but nothing works. I tried different Regions double and triple checked everything in Azure and even created Resource/Service separatly. Also I was able to get an auth token back in powershell with the Azure key so I think the problem comes from TTS.

Is there anything known yet? 

For reference, this is how I checked if it works in powershell:

$FetchTokenHeader = @{
  'Content-type'='application/x-www-form-urlencoded'
  'Content-Length'= '0'
  'Ocp-Apim-Subscription-Key' = 'KEY_HERE'
}
$OAuthToken = Invoke-RestMethod -Method POST -Uri https://germanywestcentral.api.cognitive.microsoft.com/sts/v1.0/issueToken -Headers $FetchTokenHeader
$OAuthToken

**Comment by luc-languagetools** [comment]
your screenshot exposes your API key so I would suggest re-generating it and deleting your screenshot.
in what region is your resource group ? (as per the instructions https://github.com/AwesomeTTS/awesometts-anki-addon/wiki/Microsoft-Azure-Text-to-Speech)

**Comment by JosuaWelss** [comment]
I instantly regenerated the key. Just needed to use one to get that error message. 

For the region of the resource group I tried Germany West Central and East US. (Will try again just to make sure I did not fail that anytime.

**Comment by luc-languagetools** [comment]
Sorry i don't understand, did you get it working or not ?

**Comment by JosuaWelss** [comment]
Yeah I got it working. My mistake was that after azure worked (I had problem with the student membership) I put in the key in the config for AwesomeTTS Plus (as shown in the screenshot) - which gave only the message "Key not valid". I simply did not read the Part that this field is for AwesomeTTS Plus. They key worked but in that field it expects the key that you get if you have AwesomeTTS Plus. 

TLDR: I misinterpreted the error message in the conig window where I put in the key cause I did not know it is not the same field as when creating cards.

**Comment by luc-languagetools** [comment]
Got it. Note that the configuration is a lot more straightforward in HyperTTS: https://ankiweb.net/shared/info/111623432
